### PR TITLE
Type-check enkf_main and enkf_fs_manager

### PR DESCRIPTION
**Pull request by dafeda**
## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: .mypy-strict.ini [hunk 1]
```diff
@@ -68,4 +84,4 @@ ignore_missing_imports = True
 ignore_missing_imports = True
 
 [mypy-ert.ensemble_evaluator.*]
-ignore_errors = True
+ignore_errors = True
```
Comment (pinkwah) [inline .mypy-strict.ini:87]: Unnecessary removal of newline

File: .mypy.ini [hunk 2]
```diff
@@ -128,4 +125,4 @@ ignore_missing_imports = True
 ignore_missing_imports = True
 
 [mypy-ruamel]
-ignore_missing_imports = True
+ignore_missing_imports = True
```
Comment (pinkwah) [inline .mypy.ini:128]: Ditto

File: .mypy.ini [hunk 3]
```diff
@@ -77,7 +75,6 @@ ignore_errors = True
 ignore_missing_imports = True
 ignore_errors = True
 
-
```
Comment (pinkwah) [inline .mypy.ini:80]: Basically just verify that this file is formatted okayishly

File: src/ert/_c_wrappers/enkf/config/gen_kw_config.py [hunk 4]
```diff
@@ -82,7 +84,7 @@ def __init__(
         self._set_template_file(template_file)
         self.__str__ = self.__repr__
 
-    def getTemplateFile(self):
+    def getTemplateFile(self) -> os.PathLike:
         path = self._get_template_file()
```
Comment (pinkwah) [inline src/ert/_c_wrappers/enkf/config/gen_kw_config.py:88]: `os.PathLike` is hella weird. In Python, that is the name of the type, but for whatever reason, in mypy context, it's a generic that takes in the type of the underlying string. If this is failing, do `os.PathLike[str]` (you need to have `from __future__ import annotations` for this, which you do in this file)

File: src/ert/_c_wrappers/enkf/config/gen_kw_config.py [hunk 5]
```diff
@@ -155,13 +157,13 @@ def get_priors(self) -> List["PriorDict"]:
...
 
-    def transform(self, index, value):
+    def transform(self, index: int, value: float) -> float:
         return self._transform(index, value)
 
     @staticmethod
     def values_from_files(
         realizations: List[int], name_format: str, keys: List[str]
-    ) -> npt.ArrayLike:
+    ) -> npt.NDArray:
```
Comment (pinkwah) [inline src/ert/_c_wrappers/enkf/config/gen_kw_config.py:166]: I might be wrong but I think you need to add the `dtype`, ie `npt.NDArray[np.float64]`.

File: src/ert/_c_wrappers/enkf/config/gen_kw_config.py (hunk 4 above)
Comment (dafeda) [reply src/ert/_c_wrappers/enkf/config/gen_kw_config.py:88]: Thanks, I added [str]

**Review by pinkwah — APPROVED** [review]

**Pull request merged**